packages/core/src/bus/Event.spec.ts

```diff
@@ -0,0 +1,42 @@
+import { EventDispatcher } from './EventDispatcher';
+import { Event } from './Event';
+import { instance, mock, reset, verify, when } from 'ts-mockito';
+
+class TestEvent extends Event<string> {
+  constructor(
+    payload: string,
+    type?: string,
+    correlationId?: string,
+    createdAt?: Date
```

Comment:
I don't see any obvious reason to have such a huge constructor. You don't use these params
```suggestion
    payload: string
```